pkg/config/cos.go

```diff
@@ -325,49 +303,57 @@ func SaveOriginalNetworkConfig() error {
 	return err
 }
 
-// UpdateNetworkConfig updates a cOS config stage to include steps that:
+// UpdateManagementInterfaceConfig updates a cOS config stage to include steps that:
 // - generates wicked interface files (`/etc/sysconfig/network/ifcfg-*` and `ifroute-*`)
 // - manipulates nameservers in `/etc/resolv.conf`.
 // - call `wicked ifreload all` if `run` flag is true.
-func UpdateNetworkConfig(stage *yipSchema.Stage, networks map[string]Network, run bool) error {
-	mgmtNetwork, ok := networks[MgmtInterfaceName]
-	if !ok {
-		return errors.New("no management network defined")
+func UpdateManagementInterfaceConfig(stage *yipSchema.Stage, mgmtInterface Network, run bool) (string, error) {
+	if len(mgmtInterface.Interfaces) == 0 {
+		return "", errors.New("no slave defined for management network bond")
 	}
-	if len(mgmtNetwork.Interfaces) == 0 {
-		return errors.New("no slave defined for management network bond")
+
+	switch mgmtInterface.Method {
+	case NetworkMethodDHCP, NetworkMethodStatic, NetworkMethodNone:
+	default:
+		return "", fmt.Errorf("unsupported network method %s", mgmtInterface.Method)
 	}
 
-	for name, network := range networks {
-		switch network.Method {
-		case NetworkMethodDHCP, NetworkMethodStatic, NetworkMethodNone:
-		default:
-			return fmt.Errorf("unsupported network method %s", network.Method)
-		}
+	var err error
 
-		var err error
-		if len(network.Interfaces) > 0 {
-			err = updateBond(stage, name, &network)
-		} else {
-			err = updateNIC(stage, name, &network)
+	if len(mgmtInterface.Interfaces) > 0 {
+		bondMgmt := Network{
+			Interfaces:  mgmtInterface.Interfaces,
+			Method:      NetworkMethodNone,
+			BondOptions: mgmtInterface.BondOptions,
+			MTU:         mgmtInterface.MTU,
 		}
+		err = updateBond(stage, MgmtBondInterfaceName, &bondMgmt)
 		if err != nil {
-			return err
+			return "", err
 		}
+	}
 
-		switch network.Method {
-		case NetworkMethodStatic:
-			// default gateway for static mode
-			stage.Files = append(stage.Files, yipSchema.File{
-				Path:        fmt.Sprintf("/etc/sysconfig/network/ifroute-%s", name),
-				Content:     fmt.Sprintf("default %s - %s\n", network.Gateway, name),
-				Permissions: 0600,
-				Owner:       0,
-				Group:       0,
-			})
-		case NetworkMethodDHCP, NetworkMethodNone:
-			stage.Commands = append(stage.Commands, fmt.Sprintf("rm -f /etc/sysconfig/network/ifroute-%s", name))
-		}
+	if err = updateBridge(stage, MgmtInterfaceName, &mgmtInterface); err != nil {
+		return "", err
+	}
+
+	name := MgmtInterfaceName
+	if mgmtInterface.VlanID >= 2 && mgmtInterface.VlanID <= 4094 {
+		name = fmt.Sprintf("%s.%d", name, mgmtInterface.VlanID)
+	}
+
+	switch mgmtInterface.Method {
```

Comment:
What about merging the 315 line switch expression with this switch?

Reply:
We need the mgmt name after updateBridge and updateBond.
So we cannot merge it.